src/app/public/modules/split-view/split-view.component.ts

```diff
@@ -0,0 +1,423 @@
+        this.isListVisible = true;
+        this.changeDetectorRef.markForCheck();
+      }));
+    }
+  }
+
+  @HostListener('window:resize', ['$event'])
+  public onWindowResize(event: any): void {
+    // If window size is smaller than listWidth + tolerance, shrink listWidth.
+    if (this.isListVisible && event.target.innerWidth < this.listWidth + this.widthTolerance) {
+      this.listWidth = event.target.innerWidth - this.widthTolerance;
+    }
+    this.updateBreakpoints();
+  }
+
+  public onWorkspaceEnterComplete(): void {
+    this.animationComplete.next();
+  }
+
+  public onIteratorNextButtonClick(): void {
+    this.iteratorNextButtonClick.emit();
+  }
+
+  public onIteratorPreviousButtonClick(): void {
+    this.iteratorPreviousButtonClick.emit();
+  }
+
+  private setListViewMaxWidth(): void {
+    this.listWidthMax = this.skyWindow.nativeWindow.innerWidth - this.widthTolerance;
+    setTimeout(() => {
+      this.changeDetectorRef.markForCheck();
+    });
+  }
+
+  private handleIncomingMessages(message: SkySplitViewMessage) {
+    /* tslint:disable-next-line:switch-default */
+    switch (message.type) {
+      case SkySplitViewMessageType.FocusWorkspace:
+        // If mobile, wait until animation is complete then set focus on workspace panel.
+        // Otherwise, just set focus right away.
+        if (this.isMobile) {
+          this.isListVisible = false;
+          this.animationComplete
+            .take(1)
+            .subscribe(() => {
+              this.applyAutofocus();
+            });
+        } else {
+          this.applyAutofocus();
+        }
+        this.changeDetectorRef.markForCheck();
+        break;
+
+      case SkySplitViewMessageType.IteratorDisableNextButton:
+        this.nextButtonDisabled = true;
+        this.changeDetectorRef.markForCheck();
+        break;
+
+      case SkySplitViewMessageType.IteratorDisablePreviousButton:
+        this.previousButtonDisabled = true;
+        this.changeDetectorRef.markForCheck();
+        break;
+
+      case SkySplitViewMessageType.IteratorEnableNextButton:
+        this.nextButtonDisabled = false;
+        this.changeDetectorRef.markForCheck();
+        break;
+
+      case SkySplitViewMessageType.IteratorEnablePreviousButton:
+        this.previousButtonDisabled = false;
+        this.changeDetectorRef.markForCheck();
+        break;
+    }
+  }
+
+  private updateBreakpoints(): void {
+    // Update list component.
+    if (this.listComponent) {
+      this.listComponent.updateBreakpoint(this.listWidth);
+    }
+
+    // Update workspace component.
+    const workspaceParent = this.workspaceComponentRef.nativeElement.parentElement;
+    if (this.workspaceComponent && workspaceParent) {
+      this.workspaceComponent.updateBreakpoint(workspaceParent.clientWidth);
+    }
+  }
+
+  private applyAutofocus(): void {
+    const applyAutoFocus = this.coreAdapterService.applyAutoFocus(this.elementRef);
```

Comment:
The name of the "core adapter" method `applyAutoFocus` is confusing because it's sharing the same name as this class's `applyAutoFocus` method (the core adapter method is returning a boolean, but this method is returning void). Initially, I thought `SkyCoreAdapterService.applyAutoFocus` meant you were applying focus to an element, but it's really checking if the element _has_ an input with an `autofocus` attribute. Probably too late to fix the method name (I see it was added recently), but something we might need to fix in a breaking change (something like `containsAutofocusElement` would make more sense).

Also, the `getFocusableChildrenAndApplyFocus` violates the "no use of the word 'and' in a method name". We should strive to only run one task per method.